### PR TITLE
cataclysm-dda-git: 2022-08-20 -> 2023-08-09-1243

### PR DESCRIPTION
**Pull request by rardiol**
## Description of changes

Using the official experimental build tag. 

https://github.com/rardiol/nixpkgs/pull/new/cataclysm-dda-git

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/games/cataclysm-dda/git.nix
Comment (DeeUnderscore) [inline pkgs/games/cataclysm-dda/git.nix:27]: This could be named something like `locale-patch-git.patch` to make it clearer.

**Review by DeeUnderscore — CHANGES_REQUESTED** [review]
This currently sets `VERSION` to `git-2023-08-09-1243-cdda-exp`.

Upstream sets the version string in experimentals to just the short git hash, but we can probably get away with setting `VERSION` directly to `version` ([here](https://github.com/NixOS/nixpkgs/blob/0bde9c606262d80cce97802886dfe89d78c0c813/pkgs/games/cataclysm-dda/git.nix#L31)), without the `git` and `cdda-exp` bits.

**Comment by DeeUnderscore** [comment]
Thank you for the fixes.

It looks like [x86_64-darwin is failing now](https://logs.ofborg.org/?attempt_id=3ba95265-65cc-4139-b537-77dd448b2627&key=nixos%2Fnixpkgs.248163). I am unfortunately not familiar with the Mac stuff here. We do set `OSX_MIN`, which [upstream dropped from the Makefile](https://github.com/CleverRaven/Cataclysm-DDA/commit/ad52638d9e35fc05a5b379cc5aaf1a165fae4a39), but I'm not sure if that's connected, or if Nixpkgs just tries too build against a too-old OS version in general.

**Comment by rardiol** [comment]
Not familiar with Mac either. This thread may be relevant and includes some possible solutions: https://github.com/gulrak/filesystem/issues/155

**Comment by rardiol** [comment]
@ofborg build cataclysm-dda-git

It seems it compiled now, but ofborg gives up after one hour.

**Comment by rardiol** [comment]
Seems compilation takes too long for ofborg. Since it is experimental I assume anyone running it on mac can come here ask for help if it is broken.

**Comment by DeeUnderscore** [comment]
Well, I have no further comment on the x86_64-linux side. Hopefully someone with more expertise in the darwin side can have a look, because experimental does get built in Hydra.

**Comment by cizra** [comment]
What'd it take to get this merged? Do we need to find and ping a maintainer with merge permissions?

**Comment by DeeUnderscore** [comment]
> What'd it take to get this merged? Do we need to find and ping a maintainer with merge permissions?

You could try asking for a review/merge on Discourse or Matrix, though if this breaks the Darwin build, then it should probably be either fixed or marked broken before merge.

**Comment by wegank** [comment]
Closed by #330693.

**Pull request closed without merging**
